adobe_vipm/adobe/mixins/order.py

```diff
@@ -511,15 +547,57 @@ def _create_return_order_base(
         response.raise_for_status()
         return response.json()
 
-    def _get_preview_order_line_item(self, line: dict, quantity: int) -> dict:
-        adobe_base_sku = line["item"]["externalIds"]["vendor"]
-        product_sku = get_adobe_product_by_marketplace_sku(adobe_base_sku).sku
-
-        return {
+    def _get_preview_order(
+        self, authorization: Authorization, adobe_customer_id, payload: dict
+    ) -> dict:
+        while True:
+            headers = self._get_headers(authorization)
+            response = requests.post(
+                urljoin(
+                    self._config.api_base_url,
+                    f"/v3/customers/{adobe_customer_id}/orders",
+                ),
+                params={"fetch-price": "true"},
+                headers=headers,
+                json=payload,
+                timeout=self._TIMEOUT,
+            )
+            response.raise_for_status()
+            response_json = response.json()
+            failed_discount_codes = {
+                flex_discount["code"]
+                for line_item in response_json["lineItems"]
+                for flex_discount in line_item.get("flexDiscounts", [])  # noqa: WPS441
```

Comment:
Same here, I don't think it is a good practice to use list/dict/set comprehension with internal loop :-(